core/src/test/scala/schema/provider/MetadataSchemaProviderSpec.scala

```diff
@@ -110,9 +101,41 @@ class MetadataSchemaProviderSpec
 
       Await.result(MetadataSchemaProvider.metadataSchemaProvider[Future].schemaDefinition.recover {
         case _ => Seq.empty
-      }, scala.concurrent.duration.Duration.Inf) isEqualTo Seq.empty
+      }, scala.concurrent.duration.Duration.Inf) shouldBe Seq.empty
+    }
+
+    "clusterProvider" should {
+
+      "return an error if the cluster configuration is not valid" in {
+        val reader: Reader = new StringReader("cluster = {}")
+
+        val clusterProvider = MetadataSchemaProvider.clusterProvider[Future](reader)
+
+        Await.result(clusterProvider.recover {
+          case _ => Seq.empty
+        }, scala.concurrent.duration.Duration.Inf) shouldBe Seq.empty
+      }
+
+      "return the valid configuration" in {
+        val reader: Reader = new StringReader(s"cluster = ${validClusterConfiguration.print}")
+
+        val cluster = Await.result(
+          MetadataSchemaProvider.clusterProvider[Future](reader),
+          scala.concurrent.duration.Duration.Inf)
+
+        Option(cluster.getClusterName) shouldBe validClusterConfiguration.name
+      }
+
     }
 
+    "create a metadataSchemaProvider from a Reader with the configuration" in {
+      val reader: Reader = new StringReader("cluster = {}")
```

Comment:
Can we add more test cases for different meta-schemas? makes sense?

Reply:
I didn't take a look to the coverage, let me iterate over it to see what can I add

Reply:
The problem with different configurations is that the validation is made by the builder in the Datastax driver so we'll not be testing our code. Also, I think it's well covered right now. The only missing parts are untestable code due to final, private Java classes restrictions.

[See MetadataSchemaProvider.scala coverage](https://codecov.io/gh/frees-io/freestyle-cassandra/pull/75/src/core/src/main/scala/schema/provider/MetadataSchemaProvider.scala?before=core/src/main/scala/schema/provider/MetadataSchemaProvider.scala)